### ISSUE TITLE
Wrong doc.comment on undocumented overridden declarations

### ISSUE THREAD

**Issue by johnfairh** [jpsim/SourceKitten#368]
Latest code; Xcode 832.  This goes wrong:

```swift
protocol P {
    /// docs for P.m
    func m()
}

/// docs for S
struct S: P {
    func m() {}
}
```

SourceKitten reports *doc.comment* for S.m as "docs for S".

This happens because SourceKit produces *full_as_xml* for S.m from inherited P.m.  So `SyntaxMap.commentRange(beforeOffset:)` scans back and finds the "docs for S" comment.

Would you consider accepting a PR to add some code along the lines of that removed by 114abc1 to detect this scenario?

(I do not think this is related to #269: the *full_as_xml* being present is enough to trigger.)

**Comment by jpsim** [comment]
As long as we don't regress in our docs detection in other scenarios, yes I'd gladly consider any PR 😄

**Comment by jpsim** [comment]
Fixed in #375.
